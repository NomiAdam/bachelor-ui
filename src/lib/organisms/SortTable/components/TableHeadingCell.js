import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { IoIosArrowDown, IoIosArrowUp } from 'react-icons/lib/io';

const Icon = styled.div`
    padding: 0 10px;
    visibility: ${ ({ shouldDisplay }) => (shouldDisplay ? 'visible' : 'hidden') };
`;

const StyledArrowDown = Icon.withComponent(IoIosArrowDown);
const StyledArrowUp = Icon.withComponent(IoIosArrowUp);

const StyledHeadingTableCell = styled.th`
    max-width: 100%;
	padding: 20px;
	text-align: left;
	${ ({ editable }) => (editable ? 'max-width: 100px' : '') };
`;

const TableHeadingCell = ({
    label, onRequestSort, order, beingSorted, editable,
}) => (
    <StyledHeadingTableCell onClick={ onRequestSort } editable={ editable }>
        {label}
        {order === 'asc'
            ? <StyledArrowDown shouldDisplay={ beingSorted } />
            : <StyledArrowUp shouldDisplay={ beingSorted } />
        }
    </StyledHeadingTableCell>
);

TableHeadingCell.propTypes = {
    /**
     * Number representation space between Cells
     */
    colSpan: PropTypes.number,
    /**
     * Function handler when requesting sort by given ID (key)
     */
    onRequestSort: PropTypes.func,
    /**
     * String representation by which we are sorting, either ascending or descending
     */
    order: PropTypes.string,
    /**
     * String value of cell
     */
    label: PropTypes.string,
    /**
     * Whether we are sorting by this Cell ID (key)
     */
    beingSorted: PropTypes.bool,
    /**
     * Whether cell is editable or not
     */
    editable: PropTypes.bool,
};

TableHeadingCell.defaultProps = {
    label: '',
    beingSorted: false,
};

export { StyledHeadingTableCell };
export default TableHeadingCell;
