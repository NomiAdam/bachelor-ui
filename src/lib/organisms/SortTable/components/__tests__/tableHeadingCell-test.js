/* eslint-disable no-undef */
import React from 'react';
import {shallow} from 'enzyme';
import TableHeadingCell, {StyledHeadingTableCell} from '../TableHeadingCell';

const mockRequestSort = jest.fn();

describe('TableHeadingCell', () => {
    it('should be defined', () => {
        expect(shallow(<TableHeadingCell/>)).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<TableHeadingCell/>)).toMatchSnapshot();
    });
    it('should match the snapshot when beingSorted and order is "asc" ', () => {
        expect(shallow(<TableHeadingCell beingSorted order={'asc'}/>)).toMatchSnapshot();
    });
    it('should match the snapshot when beingSorted and order is "desc" ', () => {
        expect(shallow(<TableHeadingCell beingSorted order={'desc'} />)).toMatchSnapshot();
    });
    it('should be of type StyledHeadingTableCell', () => {
        expect(shallow(<TableHeadingCell/>).type()).toBe(StyledHeadingTableCell);
    });
    it('should not trigger the action on render', () => {
        shallow(<TableHeadingCell onRequestSort={mockRequestSort} />);
        expect(mockRequestSort).not.toHaveBeenCalled();
    });
    it('should trigger the action on click', () => {
        const wrapper = shallow(<TableHeadingCell onRequestSort={mockRequestSort} />);
        wrapper.find(StyledHeadingTableCell).simulate('click');
        expect(mockRequestSort).toHaveBeenCalled();
    });
});
