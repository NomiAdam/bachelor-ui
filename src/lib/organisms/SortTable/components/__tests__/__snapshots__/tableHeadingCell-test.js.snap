// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeadingCell should match the snapshot 1`] = `
<styled.th>
  <Styled(IoIosArrowUp)
    shouldDisplay={false}
  />
</styled.th>
`;

exports[`TableHeadingCell should match the snapshot when beingSorted and order is "asc"  1`] = `
<styled.th>
  <Styled(IoIosArrowDown)
    shouldDisplay={true}
  />
</styled.th>
`;

exports[`TableHeadingCell should match the snapshot when beingSorted and order is "desc"  1`] = `
<styled.th>
  <Styled(IoIosArrowUp)
    shouldDisplay={true}
  />
</styled.th>
`;
